File: data_structures/prefix_sum/python/prefix_sum.py
from typing import List

def build_prefix_sum(a: List[int]) -> List[int]:
    n = len(a)
    pref = [0] * (n+1)
    for i in range(n):
        pref[i+1] = pref[i] + a[i]
    return pref

def query_sum(pref: List[int], l: int, r: int) -> int:
    return pref[r+1] - pref[l]

File: data_structures/prefix_sum/cpp/prefix_sum.cpp
#include <vector>

using namespace std;

vector<long long> build_prefix_sum(const vector<int>& a) {
    int n = a.size();
    vector<long long> pref(n+1, 0);
    for (int i = 0; i < n; ++i) {
        pref[i+1] = pref[i] + a[i];
    }
    return pref;
}

long long query_sum(const vector<long long>& pref, int l, int r) {
    return pref[r+1] - pref[l];
}